.github/workflows/ci.yml

```diff
@@ -46,6 +47,18 @@ jobs:
       - name: Install NPM dependencies
         run: npm install
 
+      - name: Install Spago dependencies
+        run: spago install
+
+      - name: Install Pulp dependencies
+        run: bower install
+
+      - name: Build the project
+        run: npm run build:pulp
+
+      - name: Run tests
+        run: npm run test:pulp
```

Comment:
Unfortunately `pulp` and `bower` are not available in the CI environment. Also, note that not all packages have a `package.json` file (only ones that have FFI will have one).

Possibly the easiest way to add Pulp and Bower to CI is to just add a single new stanza:

```yml
- name: Verify Pulp & Bower
  run: |
    npm install bower pulp
    npx bower install
    npx pulp build
    npx pulp test
```

This could potentially be run at the very end.

Reply:
@JordanMartinez as far as scripting can you just append this to the end of the build step? I think that’s all it would take.

Reply:
It duplicates the npm install, but that’s fine.

Reply:
@thomashoneyman I can do what I'm currently doing for the `.eslintrc.json` files. Since many of these repos use the same file, I can take one, update it to another, and calculate a diff. If I come across one file whose diff matches the expected one, than I can use that expected one.

So, could you edit the `ci.yml` file and make it exactly what you want it to be? Then I could update my script to do this across all repos that have the same initial file.

Reply:
For example, see https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/src/bash/lib/updateEslint.sh and the files here https://github.com/JordanMartinez/purescript-ecosystem-update/tree/master/files/eslint

Reply:
Sure, you can see an example here:

https://github.com/purescript-contrib/purescript-affjax/pull/170

I'm just sticking the new stanza on the end, so it should work no matter the input file. Not all of these `ci.yml` files are going to be the same.

Reply:
That solves one problem, but doesn't solve another, which is editing the `ci.yml` file to insert `purescript: "stable"` is problematic due to its inconsistency.

Reply:
Every `ci.yml` file at least has a `purs-tidy: "latest"` line, right? From when we added that across the board? So you could grep for that line and insert `purescript: "unstable"` above it.

Reply:
Ah, I hadn't thought of that. Yes, that should work.

`core` libraries don't have this, so the change I'd make in my ecosystem update script wouldn't work, but we're passed that point. And assuming we do add tidy support there, then they would be consistent like this.